x/bsonx/document_test.go

```diff
@@ -65,11 +65,20 @@ func TestDocument(t *testing.T) {
 		t.Parallel()
 		t.Run("UnmarshalingError", func(t *testing.T) {
 			t.Parallel()
-			invalid := []byte{0x01, 0x02}
-			want := bsoncore.NewInsufficientBytesError(nil, nil)
-			_, got := ReadDoc(invalid)
-			if !compareErrors(got, want) {
-				t.Errorf("Expected errors to match. got %v; want %v", got, want)
+			testCases := []struct {
```

Comment:
Small nit: we prefer to have a `name` field in our table tests to make it easy to run individual sub-tests from the command line.